plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/backup/MachineBackupManager.java

```diff
@@ -131,6 +177,13 @@ public void restoreWorkspaceBackup(final String workspaceId,
                                        final String userId,
                                        final String groupId,
                                        final String destAddress) throws ServerException {
+        ReentrantLock lock = new ReentrantLock();
+        lock.lock();
+        if (workspacesBackupLocks.putIfAbsent(workspaceId, lock) != null) {
+            LOG.error("Attempt to start backup/restore process on {} workspace while it under restore.", workspaceId);
+            return; // we cannot restore workspace while it under backup/restore process
```

Comment:
Comment is confusing a bit. Please rephrase it to explain that this code identify multiple restores of one workspace which is illegal.